umbral/kfrags.py

```diff
@@ -0,0 +1,192 @@
+"""
+Copyright (C) 2018 NuCypher
+
+This file is part of pyUmbral.
+
+pyUmbral is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+pyUmbral is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with pyUmbral. If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import hmac
+from typing import Optional
+
+from bytestring_splitter import BytestringSplitter
+
+from umbral.config import default_curve, default_params
+from umbral.curvebn import CurveBN
+from umbral.keys import UmbralPublicKey
+from umbral.point import Point
+from umbral.signing import Signature
+from umbral.params import UmbralParameters
+from umbral.curve import Curve
+
+from constant_sorrow.constants import NO_KEY, DELEGATING_ONLY, RECEIVING_ONLY, DELEGATING_AND_RECEIVING
+
+NO_KEY(b'\x00')
+DELEGATING_ONLY(b'\x01')
+RECEIVING_ONLY(b'\x02')
+DELEGATING_AND_RECEIVING(b'\x03')
+
```

Comment:
Why are you representing each of these as one byte?  Is this just for space-saving?